types/jest/index.d.ts

```diff
@@ -538,16 +538,16 @@ declare namespace jest {
         mock: MockContext<T>;
         mockClear(): void;
         mockReset(): void;
-        mockImplementation(fn: (...args: any[]) => any): Mock<T>;
-        mockImplementationOnce(fn: (...args: any[]) => any): Mock<T>;
+        mockImplementation(fn: (...args: any[]) => ReturnType<T>): Mock<T>;
```

Comment:
I'm not sure of where all `MockInstance` is being used, so I may be a little off-base here.

Isn't `T` here the function that we are trying to mock? Could this not just be:
```typescript
mockImplementation(fn: T): Mock<T>;
```

Reply:
@andnp Thanks for the comment! And yeah, I think you're right. I was just trying to do a minimal change. I'm happy to add that though.

Reply:
True, it might be out of scope for this PR.